gateway/knn/knn.go

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package knn
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"odfe-cli/client"
+	"odfe-cli/entity"
+	gw "odfe-cli/gateway"
+)
+
+const (
+	baseURL              = "_opendistro/_knn"
+	statsURL             = baseURL + "/stats"
```

Comment:
Is this case covered: A user doesnt pass in a list nodes, but does pass a list of stats?

Reply:
Thanks for pointing it out. I will update the code. Just curious, any reason why this was not considered as query parameter instead of url path parameter?

Reply:
When I wrote it, I followed the convention set by a few other plugins. Not sure if one or the other would be better design.

Reply:
Got it. I saw some elasticsearch core plugin api as well. They follow this pattern to filter list. It is weird since REST doesn't follow this pattern.